xDrip/Components/Charts/SupportingLogic/GlucoseChartProvider.swift

```diff
@@ -7,29 +7,54 @@
 //
 
 import UIKit
+import AKUtils
 
 protocol GlucoseChartProvider {
     var dateInterval: DateInterval { get }
     var yRange: ClosedRange<Double> { get }
     var entries: [GlucoseChartGlucoseEntry] { get }
+    var basalEntries: [BasalChartEntry] { get }
     var insets: UIEdgeInsets { get }
     var circleSide: CGFloat { get }
     
     func drawGlucoseChart()
 }
 
+private struct BasalEntry: BasalChartEntry {
+    var value: Double
+    var date: Date
+}
+
+// swiftlint:disable function_body_length
+
 extension GlucoseChartProvider where Self: UIView {
+    private var minDate: TimeInterval {
+        return dateInterval.start.timeIntervalSince1970
+    }
+    
+    private var maxDate: TimeInterval {
+        return dateInterval.end.timeIntervalSince1970
+    }
+    
+    private var timeInterval: TimeInterval {
+        return maxDate - minDate
+    }
+    
+    private var pixelsPerSecond: Double {
+        return Double(bounds.width - insets.left - insets.right) / timeInterval
+    }
+    
+    private var yInterval: Double {
+        return Double(bounds.height - insets.bottom - insets.top)
+    }
+    
+    private var pixelsPerValue: Double {
+        return yInterval / (yRange.upperBound - yRange.lowerBound)
+    }
```

Comment:
Those values were calculated only once before your change. Now they are calculated variables and are calculated each time they are called, which is not effective. I would suggest changing them to normal variables and adding some sort of `calculateValues()` method, something like:
```
func calculateValues() {
  minDate = dateInterval.start.timeIntervalSince1970
  pixelsPerValue = yInterval / (yRange.upperBound - yRange.lowerBound)
  // etc
}
```
and call it prior to drawing the chart.